### DISCUSSION TITLE
No longer functional

### DISCUSSION THREAD

**Discussion by industrial64** [netbox-community/netbox-docker#756] [General]
Tried bringing this up multiple times - assuming it is abandoned. Netbox never comes up on any exposed port after successful init. 
Suggest re-writing as deprecated in the instructions.

```netbox_1               | ✅ Initialisation is done.
netbox_1               | ⏳ Waiting for control socket to be created... (1/10)
netbox_1               | 2022/05/04 18:53:53 [warn] 6#6 Unit is running unprivileged, then it cannot use arbitrary user and group.
netbox_1               | 2022/05/04 18:53:53 [info] 6#6 unit started
netbox_1               | 2022/05/04 18:53:53 [info] 20#20 discovery started
netbox_1               | 2022/05/04 18:53:53 [notice] 20#20 module: python 3.9.5 "/usr/lib/unit/modules/python3.unit.so"
netbox_1               | 2022/05/04 18:53:53 [info] 6#6 controller started
netbox_1               | 2022/05/04 18:53:53 [info] 22#22 router started
netbox_1               | 2022/05/04 18:53:53 [notice] 6#6 process 20 exited with code 0
netbox_1               | 2022/05/04 18:53:53 [info] 22#22 OpenSSL 1.1.1n  15 Mar 2022, 101010ef
netbox_1               | ⚙️ Applying configuration from /etc/unit/nginx-unit.json
netbox_1               | 2022/05/04 18:53:54 [info] 26#26 "netbox" application started
netbox_1               | ✅ Unit configuration loaded successfully
netbox_1               | 2022/05/04 18:53:59 [notice] 6#6 process 18 exited with code 0
```

**Comment by tobiasge** [comment]
This looks like a normal start of the container. Do you have a port mapping in your `docker-compose.override.yml` file?

**Comment by industrial64** [comment]
On first start it never came up (waited over 5mins after seeing that prior exit code 0)

I tried again cleanly and perhaps on first start it needs time more than that - there was/is a big (>2min) delay between the 'application started' and it tendering a client connection. Didn't seem like a congestion issue - but forgive my impatience either way.

```
Attaching to netbox-docker_redis-cache_1, netbox-docker_postgres_1, netbox-docker_redis_1, netbox-docker_netbox-housekeeping_1, netbox-docker_netbox-worker_1, netbox-docker_netbox_1
redis-cache_1          | 1:C 04 May 2022 19:13:35.448 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
```

```
2022/05/04 19:14:33 [warn] 6#6 Unit is running unprivileged, then it cannot use arbitrary user and group.
netbox_1               | ⏳ Waiting for control socket to be created... (1/10)
netbox_1               | 2022/05/04 19:14:33 [info] 6#6 unit started
netbox_1               | 2022/05/04 19:14:33 [info] 16#16 discovery started
netbox_1               | 2022/05/04 19:14:33 [notice] 16#16 module: python 3.9.5 "/usr/lib/unit/modules/python3.unit.so"
netbox_1               | 2022/05/04 19:14:33 [info] 6#6 controller started
netbox_1               | 2022/05/04 19:14:33 [notice] 6#6 process 16 exited with code 0
netbox_1               | 2022/05/04 19:14:33 [info] 18#18 router started
netbox_1               | 2022/05/04 19:14:33 [info] 18#18 OpenSSL 1.1.1n  15 Mar 2022, 101010ef
netbox_1               | ⚙️ Applying configuration from /etc/unit/nginx-unit.json
netbox_1               | 2022/05/04 19:14:34 [info] 22#22 "netbox" application started
netbox_1               | ✅ Unit configuration loaded successfully
netbox_1               | 2022/05/04 19:14:41 [notice] 6#6 process 14 exited with code 0
```

```
netbox_1               | 2022/05/04 19:14:59 [info] 25#25 "netbox" application started
netbox_1               | 2022/05/04 19:16:10 [info] 18#23 *5 send(27, 7FA471203E44, 13429) failed (32: Broken pipe)
netbox_1               | REDACTED - - [04/May/2022:19:16:10 +0000] "GET / HTTP/1.1" 200 77380 "-" "Mozilla/5.0 (Windows NT

**Comment by alansenairj** [comment]
▒▓  │  ~/netbox-docker │ on   release ▓▒░ docker logs 180501                                                  ░▒▓ ✔ │ at 06:08:15  ▓▒░
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
💡 Superuser Username: admin, E-Mail: admin@example.com
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
▶️  Running the startup script /opt/netbox/startup_scripts/000_users.py
▶️  Running the startup script /opt/netbox/startup_scripts/010_groups.py
▶️  Running the startup script /opt/netbox/startup_scripts/020_object_permissions.py
▶️  Running the startup script /opt/netbox/startup_scripts/030_custom_fields.py
▶️  Running the startup script /opt/netbox/startup_scripts/040_custom_links.py
▶️  Running the startup script /opt/netbox/startup_scripts/050_tags.py
▶️  Running the startup script /opt/netbox/startup_scripts/060_webhooks.py
▶️  Running the startup script /opt/netbox/startup_scripts/070_tenant_groups.py
▶️  Running the startup script /opt/netbox/startup_scripts/080_tenants.py
▶️  Running the startup script /opt/netbox/startup_scripts/090_regions.py
▶️  Running the startup script /opt/netbox/startup_scripts/110_sites.py
▶️  Running the startup script /opt/netbox/startup_scripts/120_locations.py
▶️  Running the startup script /opt/netbox/startup_scripts/130_rack_roles.py
▶️  Running the startup script /opt/netbox/startup_scripts/140_racks.py
▶️  Running the startup script /opt/netbox/startup_scripts/150_power_panels.py
▶️  Running the startup script /opt/netbox/startup_scripts/160_power_feeds.py
▶️  Running the startup script /opt/netbox/startup_scripts/170_manufacturers.py
▶️  Running the startup script /opt/netbox/startup_scripts/180_device_roles.py
▶️  Running the startup script /opt/netbox/startup_scripts/190_device_types.py
▶️  Running the startup script /opt/netbox/startup_scripts/200_devices.py
▶️  Running the startup script /opt/netbox/startup_scripts/210_dcim_interfaces.py
▶️  Running the startup script /opt/netbox/startup_scripts/220_platforms.py
▶️  Running the startup script /opt/netbox/startup_scripts/230_route_targets.py
▶️  Running the startup script /opt/netbox/startup_scripts/240_vrfs.py
▶️  Running the startup script /opt/netbox/startup_scripts/250_rirs.py
▶️  Running the startup script /opt/netbox/startup_scripts/260_asns.py
▶️  Running the startup script /opt/netbox/startup_scripts/270_aggregates.py
▶️  Running the startup script /opt/netbox/startup_scripts/280_prefix_vlan_roles.py
▶️  Running the startup script /opt/netbox/startup_scripts/290_cluster_types.py
▶️  Running the startup script /opt/netbox/startup_scripts/300_cluster_groups.py
▶️  Running the startup script /opt/netbox/startup_scripts/310_clusters.py
▶️  Running the startup script /opt/netbox/startup_scripts/320_vlan_groups.py
▶️  Running the startup script /opt/netbox/startup_scripts/330_vlans.py
▶️  Running the startup script /opt/netbox/startup_scripts/340_virtual_machines.py
▶️  Running the startup script /opt/netbox/startup_scripts/350_virtualization_interfaces.py
▶️  Running the startup script /opt/netbox/startup_scripts/360_prefixes.py
▶️  Running the startup script /opt/netbox/startup_scripts/370_ip_addresses.py
▶️  Running the startup script /opt/netbox/startup_scripts/380_primary_ips.py
▶️  Running the startup script /opt/netbox/startup_scripts/400_services.py
▶️  Running the startup script /opt/netbox/startup_scripts/420_providers.py
▶️  Running the startup script /opt/netbox/startup_scripts/440_circuit_types.py
▶️  Running the startup script /opt/netbox/startup_scripts/450_circuits.py
✅ Initialisation is done.
⏳ Waiting for control socket to be created... (1/10)
2022/05/13 09:07:03 [warn] 7#7 Unit is running unprivileged, then it cannot use arbitrary user and group.
2022/05/13 09:07:03 [info] 7#7 unit started
2022/05/13 09:07:03 [info] 16#16 discovery started
2022/05/13 09:07:03 [notice] 16#16 module: python 3.9.5 "/usr/lib/unit/modules/python3.unit.so"
2022/05/13 09:07:03 [info] 7#7 controller started
2022/05/13 09:07:03 [notice] 7#7 process 16 exited with code 0
2022/05/13 09:07:03 [info] 18#18 router started
2022/05/13 09:07:03 [info] 18#18 OpenSSL 1.1.1n  15 Mar 2022, 101010ef
⚙️ Applying configuration from /etc/unit/nginx-unit.json
2022/05/13 09:07:04 [info] 22#22 "netbox" application started
✅ Unit configuration loaded successfully
2022/05/13 09:07:05 [notice] 7#7 process 14 exited with code 0

**Reply by alansenairj** [reply]
![image](https://user-images.githubusercontent.com/20565821/168252674-be46c7ba-6e5d-4b5d-bd7d-c13b5da5e9ac.png)

**Comment by alansenairj** [comment]
![image](https://user-images.githubusercontent.com/20565821/168253131-1efb94f8-1caf-4b8a-9fa3-45f7935f7925.png)

**Comment by alansenairj** [comment]
![image](https://user-images.githubusercontent.com/20565821/168254629-1cca6ed0-8d30-4a11-a9a5-230f13ba8af6.png)

**Reply by tobiasge** [reply]
Did you change the Nginx Unit configuration? If not why are you mapping the port to 8000?
See in the example file:
```yaml
version: '3.4'
services:
  netbox:
    ports:
      - 8000:8080
```

**Reply by alansenairj** [reply]
I can´t believe! I did it because I am using port 8080. But it worked. Thanx

**Reply by tobiasge** [reply]
The first port (8000) is the "external port". That has to be unique for your host (or IP on the host). The second is the internal port where Nginx unit is providing the service. This one is isolated from your host ports by docker.

**Reply by alansenairj** [reply]
Yes. I know, thank you. 

I was pretty confused because my host has:
Portainer using port 8000:8000
Watchtower using port 8080:8080

I need to use this netbox stack and it was a bit confusing in how to change these default ports in netbox and it's stacks. 
I think I need to change host's port to 8001 in doco(...).override.yml file , for exemple, to avoid conflict with host 8000 portainer's port. 

by the way, thanxx
